### ISSUE TITLE
Extra file-wide indentation with trailing parenthesized requires-expression in requires-clause

### ISSUE THREAD

**Issue by 976866b7-fc02-4d23-a1c0-197b08fd7c0e** [llvm/llvm-project#47920]
|  |  |
| --- | --- |
| Bugzilla Link | [48576](https://llvm.org/bz48576) |
| Version | trunk |
| OS | Linux |
| CC | @mydeveloperday,@higaski |

## Extended Description 
.clang-format:
```yaml
IndentRequires: true
```

Input and expected formatted result:
```C++
template <class T>
  requires x and (requires { y; })
struct w {};

int z;
```

Actual output:
```C++
template <class T>
  requires x and(requires { y; }) struct w {};

  int z;
```

Having a trailing, parenthesized requires-expression in a requires-clause results in the rest of the file having an extra indentation level.

**Comment by JohelEGP** [comment]
Seems fixed. The "input and expected formatted result" is now the "actual output" too.

**Comment by rymiel** [comment]
Fixed by 9aab0db13fb6d21d1b70247a9b5e4cf916ee1c3a
